src/hf_hydrodata/data_model_access.py

```diff
@@ -218,6 +218,9 @@ def _get_api_headers(required=True) -> dict:
         url_security = f"{HYDRODATA_URL}/api/api_pins?pin={pin}&email={email}"
         response = requests.get(url_security, timeout=1200)
         if not response.status_code == 200:
+            if not required:
+                # The PIN is not required so it is ok that the API request returned an error.
+                return {}
```

Comment:
If the PIN isn't required can we just skip the API call all together?  It won't do much, but it'll speed things up a tiny bit and it seems uneeded.